cpp/ql/test/query-tests/Security/CWE/CWE-129/semmle/ImproperArrayIndexValidation/test1.c

```diff
@@ -63,4 +63,17 @@ void test6(int i) {
   myTable[s] = 0; // GOOD: Input is small [FALSE POSITIVE]
 }
 
-void test7(char *s) { }
\ No newline at end of file
+typedef void FILE;
+#define EOF (-1)
+
+int getc(FILE*);
+
+extern int myMaxCharTable[256];
+
+void test7(FILE* fp) {
+  int ch;
+  while ((ch = getc(fp)) != EOF) {
+    myMaxCharTable[ch] = 0; // GOOD [FALSE POSITIVE]
```

Comment:
This is good, because `getc` either returns a value which fits in an `unsigned char` or `-1`? Correct?

Reply:
Some of the alert changes on the nightly suite seem to suggest that this might be violated on some platforms?

Reply:
Oh, sorry I should have read your top comment more closely. That clarifies things.

Reply:
Are you referring to the Kamailio alerts? I think they're TPs. They come from the `EOF` check here: https://github.com/kamailio/kamailio/blob/b02c247023ea3ea0ef9753efdb04aff7d5d3bbb4/src/modules/db_text/dbt_file.c#L143, I think

Reply:
So the bound for `fgetc` is initially `[-1, 255]`, and then the `!=` check refines it to `[0, 255]`. So the `== EOF` check never holds.

Reply:
Yup, I see that now.